### PR TITLE
Add test and docs for pubsub's setTopicExpressionString

### PR DESCRIPTION
**Pull request by ttomsu**
Fixes https://github.com/spring-cloud/spring-cloud-gcp/issues/838

### CODE REVIEW

File: docs/src/main/asciidoc/pubsub.adoc [hunk 1]
```diff
@@ -159,6 +159,9 @@ include::{project-root}/docs/src/test/java/com/google/cloud/spring/autoconfigure
 
 By default, the `SimplePubSubMessageConverter` is used to convert payloads of type `byte[]`, `ByteString`, `ByteBuffer`, and `String` to Pub/Sub messages.
 
+===== Advanced Usage
```
Comment (meltsufin) [inline docs/src/main/asciidoc/pubsub.adoc:162]: Maybe something like "SpEL expression-based message routing" for the header?

File: docs/src/main/asciidoc/pubsub.adoc [hunk 2]
```diff
@@ -159,6 +159,9 @@ include::{project-root}/docs/src/test/java/com/google/cloud/spring/autoconfigure
...
+You can use message headers to route messages to different Pub/Sub topics. To do so, add the `GcpPubSubHeaders.TOPIC` key and the desired topic name as the value to the message headers. Alternatively, use `PubSubMessageHandler.setTopicExpressionString()` to use a SpEL expression to specify the topic name.
```
Comment (meltsufin) [inline docs/src/main/asciidoc/pubsub.adoc:163]: We try to stick with Spring Team's 1-sentence per line rule. :)

Comment (meltsufin) [inline docs/src/main/asciidoc/pubsub.adoc:163]: Is it possible to include a short code snippet to demonstrate?

File: spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/integration/outbound/PubSubMessageHandlerTests.java [hunk 3]
```diff
@@ -102,7 +101,22 @@ public void testPublishDynamicTopic() {
 						.build());
 		this.adapter.handleMessage(dynamicMessage);
 		verify(this.pubSubTemplate, times(1))
-			.publish(eq("dynamicTopic"),	eq("testPayload".getBytes()), isA(Map.class));
+			.publish(eq("dynamicTopic"), eq("testPayload".getBytes()), anyMap());
+	}
+
+	@Test
+	public void testSendToExpressionTopic() {
```
Comment (meltsufin) [inline spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/integration/outbound/PubSubMessageHandlerTests.java:108]: Nice test!

File: docs/src/main/asciidoc/pubsub.adoc (hunk 1 above)
Comment (ttomsu) [reply docs/src/main/asciidoc/pubsub.adoc]: Found the block in `spring-integration-pubsub` that is a better home for this, and it doesn't need a header anymore.

File: docs/src/main/asciidoc/pubsub.adoc (hunk 2 above)
Comment (ttomsu) [reply docs/src/main/asciidoc/pubsub.adoc]: Ahh, didn't know that! Duly noted.

Comment (ttomsu) [reply docs/src/main/asciidoc/pubsub.adoc]: This comment unearthed a bunch of work over the last two days ;-) Done.

File: spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/integration/outbound/PubSubMessageHandlerTests.java (hunk 3 above)
Comment (ttomsu) [reply spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/integration/outbound/PubSubMessageHandlerTests.java:108]: :t-rex:

**Review by dzou — APPROVED** [review]

**Review by meltsufin — APPROVED** [review]

**Pull request merged**